apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.YCSB/Program.cs

```diff
@@ -0,0 +1,513 @@
+            {
+                PrintUsage();
+                return false;
+            }
+
+            if (args[1] != "-P" || args.Count < 3)
+            {
+                PrintUsage();
+                return false;
+            }
+
+            s_templateFile = args[2];
+            if (!File.Exists(s_templateFile))
+            {
+                Console.Error.WriteLine($"File {s_templateFile} does not exist!");
+                PrintUsage();
+                return false;
+            }
+
+            if (args.Count % 2 != 1)
+            {
+                //wierd error, user had a -p without another value or had an extra space in a value causing it
+                //to be misread.
+                Console.Error.WriteLine("Error parsing argument values.");
+                PrintUsage();
+                return false;
+            }
+
+            for (var i = 3; i < args.Count; i += 2)
+            {
+                if (args[i] != "-p")
+                {
+                    Console.Error.WriteLine($"Expected a '-p' at argument {(i - 1) / 2}");
+                    PrintUsage();
+                    return false;
+                }
+                var keyValue = args[i + 1].Split('=');
+                if (keyValue.Length != 2)
+                {
+                    Console.Error.WriteLine($"Unable to parse key/value for arg {(i - 1) / 2}");
+                    PrintUsage();
+                    return false;
+                }
+                s_options[keyValue[0].ToLowerInvariant()] = keyValue[1];
+            }
+            s_debugMode = s_options.ContainsKey("debugmode");
+            return true;
+        }
+
+        private static void DebugMessage(string message)
+        {
+            if (s_debugMode)
+            {
+                Console.WriteLine(message);
+            }
+        }
+
+        private static void PrintUsage()
+            => Console.Error.WriteLine(
+                "Usage:\n ycsb_spanner_csharp run -P <relative_path_to_template_file> [-p property=value]...");
+
+        [SuppressMessage("ReSharper", "PossibleMultipleEnumeration")]
+        private static double Variance(IEnumerable<double> source)
+        {
+            if (!source.Any())
+            {
+                return 0;
+            }
+            double mean = source.Average();
+            return source.Select(d => (d - mean) * (d - mean)).Average();
+        }
+
+        private class Operation
+        {
+            public Operation(string proportionField, Func<ConcurrentBag<double>, Task> action)
+            {
+                ProportionField = proportionField;
+                Name = proportionField.Replace("proportion", "").ToUpperInvariant();
+                Action = action;
+            }
+
+            public ConcurrentBag<double> Latencies { get; } = new ConcurrentBag<double>();
+            public string Name { get; }
+            public double Proportion => GetOption<double>(ProportionField);
```

Comment:
What's this meant to indicate? I can see it referred to from a bunch of places, but I can't easily understand the purpose.

(After a while I think I understand it, but a comment would be really helpful.)